SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialReplicationGraph.cpp

```diff
@@ -87,3 +91,179 @@ void USpatialReplicationGraph::PostReplicateActors(UNetReplicationGraphConnectio
+
+	UNetReplicationGraphConnection* ConnectionManager =
+		Cast<UNetReplicationGraphConnection>(NetConnection->GetReplicationConnectionDriver());
+	if (ConnectionManager == nullptr)
+	{
+		UE_LOG(
+			LogSpatialReplicationGraph, Error,
+			TEXT("Replication connection driver was null when trying to calculate client interest, did we call this before replication?"))
+		return {};
+	}
+
+	FNetViewerArray Viewer;
+	Viewer.Emplace(NetConnection, 0);
+	TSet<FName> AllClientVisibleLevelNames;
+	ConnectionManager->GetClientVisibleLevelNames(AllClientVisibleLevelNames);
+
+	// Iterate across all global and connection relevant nodes and gather a (broadphase) list of client interested Actors
+	// Ideally we do interest list caching per node, and only gather new lists from nodes where interest was marked dirty.
+	FGatheredReplicationActorLists GatheredReplicationListsForConnection;
+	{
+		QUICK_SCOPE_CYCLE_COUNTER(NET_ClientEntityInterest_GatherRepListsFromNodes);
+
+		const FConnectionGatherActorListParameters ClientActorInterestParameters(Viewer, *ConnectionManager, AllClientVisibleLevelNames,
+																				 GetReplicationGraphFrame(),
+																				 GatheredReplicationListsForConnection, true);
+		for (UReplicationGraphNode* Node : GlobalGraphNodes)
+		{
+			Node->GatherClientInterestedActors(ClientActorInterestParameters);
+		}
+
+		for (UReplicationGraphNode* Node : ConnectionManager->GetConnectionGraphNodes())
+		{
+			Node->GatherClientInterestedActors(ClientActorInterestParameters);
+		}
+	}
+
+	// Parse gathered rep lists into a list of Actors to return to the InterestFactory.
+	// - remove duplicate Actors from consideration
+	// - ignore our non-spatial and dormant Actors
+	// - add dependent Actors
+	TArray<AActor*> ClientInterestedActors =
+		ExtractClientInterestActorsFromGather(ConnectionManager, GatheredReplicationListsForConnection, Viewer);
+
+	return ClientInterestedActors;
+}
+
+TArray<AActor*> USpatialReplicationGraph::ExtractClientInterestActorsFromGather(
+	UNetReplicationGraphConnection* ConnectionManager, FGatheredReplicationActorLists& GatheredReplicationListsForConnection,
+	FNetViewerArray& Viewers)
+{
+	QUICK_SCOPE_CYCLE_COUNTER(NET_ClientEntityInterest_ExtractFromGatheredLists);
+
+	TArray<AActor*> ClientInterestedActors{};
+
+	FPerConnectionActorInfoMap& ConnectionActorInfoMap = ConnectionManager->ActorInfoMap;
+	const uint32 FrameNum = GetReplicationGraphFrame();
+
+	TArray<AActor*> SortingArray{};
+	// Should try and reserve accurately here - could sum gathered rep list lengths (although this wouldn't factor dependent Actors) or
+	// cache the last list size
+
+	for (const FActorRepListConstView& List : GatheredReplicationListsForConnection.GetLists(EActorRepListTypeFlags::Default))
+	{
+		for (AActor* Actor : List)
+		{
+			if (!Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_SpatialType))
+			{
+				UE_LOG(LogSpatialReplicationGraph, Error,
```

Comment:
I guess this is an error because we should have ensured that a non spatial type actor was not added to the list elsewhere in code?

Reply:
No - but you could try and that would be a bad thing as it wouldn't replicate. The key part here is the message to update the routing table.